src/kaboom.ts

```diff
@@ -5467,11 +5477,335 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
 
 	}
 
-	function obstacle(): ObstacleComp {
+	function obstacle(isObstacle: boolean = true) : ObstacleComp {
+		let _isObstacle = isObstacle
 		return {
 			id: "obstacle",
-			require: [ "tile" ],
-			isObstacle: true,
+			require: ["tile"],
+			set isObstacle(isObstacle: boolean) {
+				if (_isObstacle !== isObstacle) {
+					_isObstacle = isObstacle;
+					(this as unknown as GameObj<ObstacleComp | TileComp>).getLevel().invalidateNavigationMap();
+				}
+			},
+			get isObstacle(): boolean {
+				return _isObstacle;
+			}
+		}
+	}
+
+	function cost(cost: number = 1) : CostComp {
+		return {
+			id: "cost",
+			require: ["tile"],
+			cost: cost,
+		}
+	}
+
+	function edges(mask: EdgeMask | string[] = EdgeMask.All) : EdgesComp {
+		const lookup = {
+			"left": EdgeMask.Left,
+			"top": EdgeMask.Top,
+			"right": EdgeMask.Right,
+			"bottom": EdgeMask.Bottom,
+		}
+
+		const _mask: EdgeMask = Array.isArray(mask) ?
+			mask.map(s => lookup[s] || 0).reduce((mask, dir) => mask | dir, 0) :
+			mask
+
+		return {
+			id: "edges",
+			require: ["tile"],
+			mask: _mask,
+		}
+	}
+
+	function pathfinding() : PathfindingComp {
+		let _costMap: number[] | undefined
+		let _edgeMap: number[] | undefined
+		let _connectivityMap: number[] | undefined
```

Comment:
Nit: Can we type them to `number[] | null` here and initialize to `null`?

Reply:
Sure